src/hmrc-design-patterns/service-timeout/index.njk

```diff
@@ -0,0 +1,200 @@
+    <li>say what will happen to their answers, if needed</li>
+    <li>have a ‘Stay signed in’ button</li>
+    <li>have a ‘Sign out’ link that signs them out as normal</li>
+  </ul>
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'default',
+    description: 'Service timeout default',
+    height: 400
+  }) }}
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'saved',
+    description: 'Saved service timeout',
+    height: 400
+  }) }}
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'not-saved',
+    description: 'Not saved service timeout',
+    height: 400
+  }) }}
+
+  <p class="govuk-body">If they do nothing, take them to a ‘We signed you out’ page that uses similar content as the warning box.</p>
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'signed-out-default',
+    description: 'Signed out timeout default'
+  }) }}
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'signed-out-saved',
+    description: 'Signed out timeout saved'
+  }) }}
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'signed-out-not-saved',
+    description: 'Signed out timeout not saved'
+  }) }}
+
+  <h3 class="govuk-heading-m" id="when-the-user-is-not-signed-in">When the user is not signed in</h3>
+
+  <p class="govuk-body">The warning should:</p>
+
+  <ul class="govuk-list govuk-list--bullet">
+    <li>say ‘For your security, we will delete your answers in 2 minutes.’</li>
+    <li>have a button to stay in the service</li>
+    <li>have a ‘Delete your answers’ link that takes them to the ‘You deleted your answers’ page</li>
+  </ul>
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'not-signed-in',
+    description: 'Not signed in page',
+    height: 400
+  }) }}
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'you-deleted-your-answers',
+    description: 'You deleted your answers page'
+  }) }}
+
+  <p class="govuk-body">If they do nothing, delete their answers and take them to the ‘For your security, we deleted your answers’ page.</p>
+
+  {{ example({
+    item: 'service-timeout',
+    example: 'we-deleted-your-answers',
+    description: 'We deleted your answers page'
+  }) }}
+
+  <h3 class="govuk-heading-m" id="technical-details">Technical details</h3>
+
+  <p class="govuk-body">Add the hmrc-timeout-dialog meta tag and initialise the timeout dialog Javascript</p>
+
+  {{
+    codeSnippet({
+      code: '<meta
+  name="hmrc-timeout-dialog"
+  data-timeout=""
+  data-countdown=""
+  data-keep-alive-url=""
+  data-sign-out-url=""
+  data-title=""
+  data-message=""
+  dat-message-suffix=""
+  data-keep-alive-button-text=""
+  data-sign-out-button-text=""
+/>'
+    })
+  }}
+
+  {{
+    codeSnippet({
+code:'timeoutDialog = HMRC.TimeoutDialog.init()'
+    })
+  }}
+
+  <ul class="govuk-list govuk-list--bullet">
+    <li>data-timeout is the number of seconds before the user times out and 900 is the platform default</li>
```

Comment:
Would it be worth documenting data-language? It's supported if using the HTML directly if not in the Nunjucks template yet.

Reply:
@mikeash82 I think this would be worth looking at as soon as the pattern's published. What do you think?